### ISSUE TITLE
crash if filter ram user name by *

### ISSUE THREAD

**Issue by y-zh** [aliyun/terraform-provider-alicloud#3506]
Try below code:

data "alicloud_ram_users" "ram_usr_test" {
  output_file = "users.txt"
  group_name = "Test"
  name_regex = "*"
}

cause crash

Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-alicloud_v1.121.2.exe plugin:

panic: regexp: Compile(`*`): error parsing regexp: missing argument to repetition operator: `*`

goroutine 644 [running]:
regexp.MustCompile(0xc0a5c70, 0x1, 0x5af2200)
        regexp/regexp.go:311 +0x15f
github.com/aliyun/terraform-provider-alicloud/alicloud.dataSourceAlicloudRamUsersRead(0xc000250af0, 0x659cfe0, 0xc000600400, 0xc000250af0, 0x0)
        github.com/aliyun/terraform-provider-alicloud/alicloud/data_source_alicloud_ram_users.go:128 +0x11a5
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).ReadDataApply(0xc000571880, 0xc000699160, 0x659cfe0, 0xc000600400, 0xc00088f4e0, 0x1, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.4.0/helper/schema/resource.go:398 +0x94
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).ReadDataApply(0xc000b62e80, 0xc001427ad0, 0xc000699160, 0xc000699160, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.4.0/helper/schema/provider.go:451 +0x96
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc000006d28, 0x6b407c0, 0xc000fe2150, 0xc00164d580, 0xc000006d28, 0xc000fe2150, 0xc000cd1b78)
        github.com/hashicorp/terraform-plugin-sdk@v1.4.0/internal/helper/plugin/grpc_provider.go:1036 +0x445
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadDataSource_Handler(0x5e91600, 0xc000006d28, 0x6b407c0, 0xc000fe2150, 0xc0016a9800, 0x0, 0x6b407c0, 0xc000fe2150, 0xc0004c2240, 0x84)
        github.com/hashicorp/terraform-plugin-sdk@v1.4.0/internal/tfplugin5/tfplugin5.pb.go:3225 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000105380, 0x6bbe400, 0xc0005c4300, 0xc00141e900, 0xc000f27980, 0xc1874b0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.27.0/server.go:1024 +0x522
google.golang.org/grpc.(*Server).handleStream(0xc000105380, 0x6bbe400, 0xc0005c4300, 0xc00141e900, 0x0)
        google.golang.org/grpc@v1.27.0/server.go:1313 +0xd54
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000566a30, 0xc000105380, 0x6bbe400, 0xc0005c4300, 0xc00141e900)
        google.golang.org/grpc@v1.27.0/server.go:722 +0xa8
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.27.0/server.go:720 +0xa8

Error: The terraform-provider-alicloud_v1.121.2.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

**Comment by xiaozhu36** [comment]
Hello @y-zh This error caused by there is a incorrect name_regex value and if you want to fetching all of users in the `Test` group, you can using the following the configurations:
```
data "alicloud_ram_users" "ram_usr_test" {
output_file = "users.txt"
group_name = "Test"
name_regex = ".*"
}
```

In addition, we will improve the error and avoid the crash error. I am sorry for it.

**Comment by xiaozhu36** [comment]
HI @y-zh It has been fixed in the version 1.122.1. Please have a check.
